conda_lock/conda_lock.py

```diff
@@ -745,13 +813,37 @@ def create_lockfile_from_spec(
         for dep in deps:
             locked[(dep.manager, dep.name, dep.platform)] = dep
 
+    git_metadata = GitMeta.create() if add_git_metadata else None
+    time_metadata = TimeMeta.create() if add_time_metadata else None
+    inputs_metadata: Optional[Dict[str, InputMeta]] = (
+        {
+            relative_path(lockfile_path.parent, src_file): InputMeta.create(
+                src_file=src_file
+            )
+            for src_file in src_files
+        }
+        if src_files is not None
+        else None
+    )
```

Comment:
I am happy to provide some input here after having fiddled around with consistency of source files in my recent PRs #230 #232

In general, we should make sure the inputs metadata are added to the lock file consistently with the existing `sources` key that conda-lock is already including in the metadata, which are passed as `sources` to `LockMeta` below

- As you can see, there is no need to have the `src_files` as an argument, since the same is available as `spec.sources`
- It is important to make sure paths are resolved in the same way in the `sources` and `inputs` metadata, which is not the case: `source.resolve()` below vs. `relative_path(lockfile_path.parent, src_file)

Something along the lines of the following might do the job

```py
def create_lockfile_from_spec(
    # [...]
    # boolean add_inputs_metadata replacing src_files
):
# [...]
    # assign resolved sources first (still with resolve(), this will be changed in #230 )
    spec_sources = {str(source.resolve()): source for source in spec.sources}
    inputs_metadata: Optional[Dict[str, InputMeta]] = (
        {
            resolved: InputMeta.create(src_file=src_file)
            for src_key, src_file in spec_sources.items()
        }
        if add_inputs_metadata
        else None
    )
    # [...]
    return Lockfile(
        package=[locked[k] for k in locked],
        metadata=LockMeta(
            # [...]
            sources=list(spec_sources.keys()),
            # [...]
```

Reply:
Hi Riccardo, previously @mariusvniekerk had indicated relative paths were preferable to him in the context of the lock-file. I'm not sure which person I should be listening to in order to get this merged, but it seems to me that `resolve` does the opposite of what @mariusvniekerk previously requested.  https://github.com/conda-incubator/conda-lock/pull/204#discussion_r933222792

Reply:
Indeed, I agree that relative paths should be the way to go, which is also the reason for me opening #229 after the absolute paths were introduced in #189, probably too "wildly" to support #175

If @mariusvniekerk agrees, feel free to use relative paths in the aligned `spec_sources` hinted above, I will then rework my open PR #230, to re-handle different drives.

Reply:
It seems like #175 becomes a problem on windows tests with my implementation. Do you have a timeframe you hope to re-address #175 in #229?

Reply:
Indeed, because the change for #175 was backed by unit tests that now fail. I would be ready anytime to change my proposed change for #229 in #230 to deal with the change you are introducing here (there is clearly a conflict between your branch and my branch).

My idea was to wait for this PR to be merged first and re-work #330 (aka rebase and solve conflicts) shortly after, but of course it is not great for your PR now with the failing test, unless @mariusvniekerk is fine to merge this despite the (motivated) failure.

One simple alternative could be the following: I propose my changes as PR to your branch in your repo, and @mariusvniekerk would integrate the fix for #229 together with your larger changes in the PR.

What do you think?

Reply:
@NatPRoach, I was probably over-thinking... the easy way to work this out is to restore passing
`sources=[str(source.resolve()) for source in spec.sources]`

```py
    # [...]
    return Lockfile(
        package=[locked[k] for k in locked],
        metadata=LockMeta(
            # [...]
            sources=[str(source.resolve()) for source in spec.sources],
            # [...]
```
This is also pretty fair since what you are proposing in this PR was never meant to change any of the pre-exiting behavior. I will re-align the two as part of my PR.

Reply:
@NatPRoach, sorry, I guess you need to take a further step back, since
```py
    spec_sources = {
        relative_path(lockfile_path.parent, source): source for source in spec.sources
    }
```
would still always be executed but does not handle yet the different drives (which again will be supported again with my PR).
We should scrap the `spec_sources` above an go for (note also the modified condition, fixing the wrong `add_inputs_metadata is not None`):
```py
    inputs_metadata: Optional[Dict[str, InputMeta]] = (
        {
            relative_path(lockfile_path.parent, src_file): InputMeta.create(
                src_file=src_file
            )
            for src_file in spec.sources
        }
        if add_inputs_metadata
        else None
    )
```
This is basically your very first code, just using `spec.sources` and replacing the argument with `add_inputs_metadata`